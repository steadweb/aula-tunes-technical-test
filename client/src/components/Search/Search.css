.search {
  margin: 30px 0;
  width: 100%;
}

select,
input {
  border: 0;
  background-color: var(--primary-color);
  color: var(--secondary-color);
  padding: 20px;
  font-size: 24px;
  border: 1px solid var(--secondary-color);
  border-radius: 10px;
  width: 500px;
}

select {
  -webkit-border-radius: 4px;
  -moz-border-radius: 4px;
  border-radius: 4px;
  background: #f8f8f8;
  color: #888;
  border: none;
  outline: none;
  display: inline-block;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  cursor: pointer;
}

@media only screen and (max-width: 600px) {
  select,
  input {
    width: calc(100% - 40px);
    margin: 0 20px;
    font-size: 20px;
  }
}
