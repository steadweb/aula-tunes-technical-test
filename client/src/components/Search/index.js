import React, { useState } from 'react';
import './Search.css';

export default function Search({ onChange, value }) {
  return (
    <div className='search'>
      <input
        type='search'
        name='search'
        placeholder='Search...'
        value={value}
        onChange={(e) => onChange(e.target.value)}
      />
    </div>
  );
}
