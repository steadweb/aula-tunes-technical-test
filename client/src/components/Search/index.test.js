import React from 'react';
import renderer from 'react-test-renderer';
import Search from './index';

describe('Search', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Search />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
