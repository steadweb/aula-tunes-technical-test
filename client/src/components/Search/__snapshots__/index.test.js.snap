// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search renders correctly 1`] = `
<div
  className="search"
>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Search..."
    type="search"
  />
</div>
`;
